Make it possible to specify suffix for temp files

elemoine opened the pull request:
# Overview

This PR suggests adding a `tempfile_suffix` argument to `ComplexInput`. This is useful when the program wrapped in a WPS Process requires specific file extensions.

# Related Issue / Discussion

See https://github.com/geopython/pywps/issues/242.

# Contribution Agreement

(as per https://github.com/geopython/pywps/blob/master/CONTRIBUTING.rst#contributions-and-licensing)

- [X] I'd like to contribute [feature X|bugfix Y|docs|something else] to PyWPS. I confirm that my contributions to PyWPS will be compatible with the PyWPS license guidelines at the time of contribution.
- [ ] I have already previously agreed to the PyWPS Contributions and Licensing Guidelines

jachym commented:
the patch is OK, simple, straight-forward. Maybe, (you knew, this will come), some tests could be included? `tests/test_inout.py:101 test_file()` ? I guess, 2 lines should fix that.

What do you think about some logic introduced to Format? Wouldn't that be more appropriate? I would find it maybe nicer - ComplexIn/Output can have more formats -> more extensions would be needed. The extension is bound to format (and mimeType).

I know, I should've told you earlier, @cehbrecht brought this topic on right time. Still, could you try to rewrite this? Sorry for inconveniences.

elemoine commented:
Thanks @jachym. Yes, I'll try using the Format extension.

elemoine commented:
@jachym this looks tricky to me. The issue is that `IOHandler` doesn't know about the data format. The data format is handled by `BasicComplex`. Did you have something in mind for the implementation of that?

elemoine commented:
@jachym a bit ugly but works: https://github.com/geopython/pywps/compare/develop...elemoine:suffix2.

elemoine commented:
@jachym I updated my branch, and added a test. I chose to go the simple path, just setting a suffix when there's a `data_format` with an extension. This may not cover all the use-cases, but it is simple and easy to change/extend in the future.
